server/autotest_server/testers/models.py

```diff
@@ -0,0 +1,215 @@
+
+
+class PyTATestDatum(BaseTestDatum, kw_only=True):
+    student_files: Annotated[List[PyTAStudentFile], Meta(title="Files to check", min_length=1)]
+    config_file_name: Optional[Annotated[FilesList, Meta(title="PyTA configuration")]] = None
+    feedback_file_names: Optional[Annotated[List[str], Meta(title="Feedback files")]] = None
+    upload_annotations: Optional[Annotated[bool, Meta(title="Upload Annotations")]] = None
+
+
+class RacketScriptFile(Struct, kw_only=True):
+    script_file: Optional[Annotated[FilesList, Meta(title="Test file")]] = None
+    test_suite_name: Optional[Annotated[str, Meta(title="Test suite name")]] = "all-tests"
+
+
+class RacketTestDatum(BaseTestDatum, kw_only=True):
+    script_files: Annotated[List[RacketScriptFile], Meta(title="Test files", min_length=1)]
+    feedback_file_names: Optional[Annotated[List[str], Meta(title="Feedback files")]] = None
+
+
+class PyTesterSchema(BaseTesterSchema, kw_only=True):
+    env_data: Annotated[PythonEnvData, Meta(title="Python environment")]
+    test_data: Annotated[List[PyTestDatum], Meta(title="Test Groups")]
+    _env: Optional[dict[str, str]] = None
+
+
+class PyTestDatum(BaseTestDatum, kw_only=True):
+    output_verbosity: Optional[Annotated[int | str, Meta(title="Output verbosity")]] = None
+    tester: Optional[Annotated[Literal["pytest", "unittest"], Meta(title="Python tester")]] = None
+
+
+class CustomTesterSchema(BaseTesterSchema, kw_only=True):
+    test_data: Optional[Annotated[List[CustomTestDatum], Meta(title="Test Groups")]] = None
+
+
+class HaskellTesterSchema(BaseTesterSchema, kw_only=True):
+    test_data: Optional[Annotated[List[HaskellTestDatum], Meta(title="Test Groups")]] = None
+
+
+class JavaTesterSchema(BaseTesterSchema, kw_only=True):
+    test_data: Optional[Annotated[List[JavaTestDatum], Meta(title="Test Groups")]] = None
+    _env: dict[str, str]
+
+
+class JupyterTesterSchema(BaseTesterSchema, kw_only=True):
+    env_data: Annotated[PythonEnvData, Meta(title="Python environment")]
+    test_data: Optional[Annotated[List[JupyterTestDatum], Meta(title="Test Groups")]] = None
+
+
+class PyTATesterSchema(BaseTesterSchema, kw_only=True):
+    env_data: Annotated[PyTAEnvData, Meta(title="Python environment")]
+    test_data: Optional[Annotated[List[PyTATestDatum], Meta(title="Test Groups")]] = None
+
+
+class RTesterSchema(BaseTesterSchema, kw_only=True):
+    env_data: Annotated[REnvData, Meta(title="R environment")]
+    test_data: Optional[Annotated[List[RTestDatum], Meta(title="Test Groups")]] = None
+
+
+class RacketTesterSchema(BaseTesterSchema, kw_only=True):
+    test_data: Optional[Annotated[List[RacketTestDatum], Meta(title="Test Groups")]] = None
+
+
+TestDatum = Union[
+    PyTestDatum,
+    HaskellTestDatum,
+    JavaTestDatum,
+    JupyterTestDatum,
+    PyTATestDatum,
+    RacketTestDatum,
+    CustomTestDatum,
+    RTestDatum,
+]
+
+TesterSchemas = Union[
```

Comment:
Unless I'm mistaken, I think type names are generally singular, even when representing a union type. So `TesterSchema` rather than `TesterSchemas`.

Similar comment above for `TestDataCategories` and `InstalledTesters`.